libraries/Microsoft.Bot.Connector/Authentication/Authenticators.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.ExceptionServices;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Connector.Authentication
+{
+    internal sealed class Authenticators : IAuthenticator
+    {
+        private readonly IAuthenticator[] inners;
+
+        public Authenticators(params IAuthenticator[] inners)
+        {
+            this.inners = inners ?? throw new ArgumentNullException(nameof(inners));
+        }
+
+        async Task<AuthenticatorResult> IAuthenticator.GetTokenAsync(bool forceRefresh)
+        {
+            ExceptionDispatchInfo info = null;
+
+            for (int index = 0; index < this.inners.Length; ++index)
```

Comment:
I can't follow this. I'm lost across "info, index, this.inners, and inners".

Please consider easier variable names.